api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceAuthService.java

```diff
@@ -53,10 +55,12 @@ public WorkspaceAuthService(
    * amount of Google Cloud computation costs (starting a notebook runtime) or increase the
    * monthly cost of the workspace (ex. creating GCS objects).
    */
-  public void validateActiveBilling(String workspaceNamespace, String workspaceId)
+  public void validateActiveBilling(
+      String workspaceNamespace, String workspaceId, WorkbenchConfig workbenchConfig)
       throws ForbiddenException {
-    if (BillingStatus.INACTIVE.equals(
-        workspaceDao.getRequired(workspaceNamespace, workspaceId).getBillingStatus())) {
+    DbWorkspace workspace = workspaceDao.getRequired(workspaceNamespace, workspaceId);
+    if (isFreeTier(workspace.getBillingAccountName(), workbenchConfig)
```

Comment:
right - we do need to add this check.  good catch.

Would it make sense to rename this method to be more explicit about this?  Maybe `validateCanUseInitialCredits()` ?

Probably should update the error text at L65 too.

Reply:
Good call